app/controllers/application_controller.rb

```diff
@@ -6,8 +6,11 @@ class ApplicationController < ActionController::Base
   before_action :set_locale
 
   rescue_from CanCan::AccessDenied do |ex|
-    flash[:error] = ex.message
-    render text: '', layout: true, status: :forbidden
+    if current_user.nil?
+      redirect_to :new_user_session, alert: ex.message
+    else
+      redirect_to :root, alert: ex.message + " Should you be? Contact webmaster."
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.